### PR TITLE
CT: Improve error messaging when there is a timeout#6662

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
The GI Bill Comparison Tool currently handles an outage or timeout of the GIDS service by endlessly spinning, so this story seeks to provide those using the CT with an error message letting them know what they need to do.

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/6662

## Testing done
Local testing completed

## Screenshots
Landing:
Screen Shot 2020-03-18 at 11 48 26 PM
Search:
Screen Shot 2020-03-18 at 11 47 35 PM

Profile:
Screen Shot 2020-03-19 at 7 27 53 AM

## Acceptance criteria
- [x] When the GI Bill Comparison Tool experiences an outage or timeout of the GIDS service, the following alert is displayed: We’re sorry. Something went wrong on our end. Please refresh this page or try again later.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/containers/GiBillApp.jsx [hunk 1]
```diff
@@ -85,7 +86,10 @@ export class GiBillApp extends React.Component {
     if (constants.inProgress) {
       content = <LoadingIndicator message="Loading..." />;
     } else {
-      content = this.props.children;
+      content =
+        constants.statusCode === (503 || 500 || 504)
```
Comment (zurbergram) [inline src/applications/gi/containers/GiBillApp.jsx]: Should this be replaced with a function in `src/applications/gi/utils/helpers.js` and then replace other usages with the function?

Comment (ksantiagomorgan6) [reply src/applications/gi/containers/GiBillApp.jsx]: its unique to each page

File: src/applications/gi/actions/index.js [hunk 2]
```diff
@@ -98,14 +101,22 @@ export function fetchConstants(version) {
...
   return dispatch => {
-    dispatch({ type: FETCH_CONSTANTS_STARTED });
-    return fetch(url, api.settings)
-      .then(res => res.json())
-      .then(
-        payload =>
-          withPreview(dispatch, { type: FETCH_CONSTANTS_SUCCEEDED, payload }),
-        err => dispatch({ type: FETCH_CONSTANTS_FAILED, err }),
-      );
+    fetch(url, api.settings)
```
Comment (shawkey-daniel) [inline src/applications/gi/actions/index.js]: Is dispatch FETCH_CONSTANTS_STARTED no longer needed?

File: src/applications/gi/containers/GiBillApp.jsx (hunk 1 above)
Comment (shawkey-daniel) [inline src/applications/gi/containers/GiBillApp.jsx]: Need descriptive constants for the error codes here and in other files, so we don't have magic numbers.

File: src/applications/gi/tests/containers/GiBillApp.unit.spec.js [hunk 3]
```diff
@@ -43,3 +43,18 @@ describe('<GiBillApp>', () => {
...
+    constants: {
+      ...defaultProps.constants,
+      statusCode: 503,
+      inProgress: true,
+    },
+  };
+  const tree = SkinDeep.shallowRender(
+    <GiBillApp {...props} location={location} params={params} />,
+  );
+  expect(tree.subTree('div')).to.be.ok;
```
Comment (shawkey-daniel) [inline src/applications/gi/tests/containers/GiBillApp.unit.spec.js]: How does this test that the error message rendered?

File: src/applications/gi/tests/containers/ProfilePage.unit.spec.js [hunk 4]
```diff
@@ -44,4 +44,21 @@ describe('<ProfilePage>', () => {
...
+        ...defaultProps.profile,
+        statusCode: 503,
+        attributes: {},
+      },
+      params: {
+        preSelectedProgram: '',
+      },
+    };
+    const tree = SkinDeep.shallowRender(<ProfilePage {...vetTecProps} />);
+    expect(tree.subTree('div')).to.be.ok;
```
Comment (shawkey-daniel) [inline src/applications/gi/tests/containers/ProfilePage.unit.spec.js]: How does this test that the error message rendered?

File: src/applications/gi/tests/containers/SearchPage.unit.spec.js [hunk 5]
```diff
@@ -27,36 +28,18 @@ describe('<SearchPage>', () => {
...
-        <SearchPage {...props} />
-      </Provider>,
-    );
-
-    expect(tree.find('LoadingIndicator').text()).to.equal(
-      'Loading search results...',
-    );
-    tree.unmount();
+    const tree = SkinDeep.shallowRender(<SearchPage {...props} />);
+    expect(tree.subTree('div')).to.be.ok;
```
Comment (shawkey-daniel) [inline src/applications/gi/tests/containers/SearchPage.unit.spec.js]: How does this test that the error message rendered?

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

**Pull request closed without merging**